### ISSUE TITLE
GL backend hangs on wayland linux

### ISSUE THREAD

**Issue by Dirbaio** [gfx-rs/gfx#3325]
When running the `quad` example with the `gl` backend on Wayland (Sway), execution hangs and no window is shown. Some "printf-debugging" shows it's hanging at the `present_surface` call.

```
RUST_LOG=trace cargo run --bin quad --features=gl
```

Full logs: https://gist.github.com/Dirbaio/a089321c6aff329ac95b14828d6f030c

Forcing to run with x11 (xwayland) works. `WAYLAND_DISPLAY= RUST_LOG=trace cargo run --bin quad --features=gl`
Vulkan backend also works correctly: `RUST_LOG=trace cargo run --bin quad --features=vulkan`

Short info header:
- GFX version: 61a01567 (latest master at the time of writing)
- OS: Linux 5.7.9-arch1-1, Sway v1.5
- GPU: Mesa Intel(R) UHD Graphics 630 (CFL GT2), Version: 4.6, (Core Profile) Mesa 20.1.4

**Comment by kvark** [comment]
Yep, GL is totally broken right now. We landed a big refactor on master, than tried to port it on hal-0.5 in #3216 and realized the whole approach was flawed. Now it's not usable, sorry!
